pkg/virt-controller/watch/network_test.go

```diff
@@ -220,6 +222,53 @@ var _ = Describe("Network interface hot{un}plug", func() {
 			),
 			!ordinal),
 	)
+
+	DescribeTable("spec interfaces",
+		func(specIfaces []v1.Interface, statusIfaces []v1.VirtualMachineInstanceNetworkInterface,
+			expectedInterfaces []v1.Interface, expectedNetworks []v1.Network) {
+			var testNetworks []v1.Network
+			for _, iface := range specIfaces {
+				testNetworks = append(testNetworks, v1.Network{Name: iface.Name})
+			}
+			testStatusIfaces := vmispec.IndexInterfacesFromStatus(statusIfaces,
+				func(i v1.VirtualMachineInstanceNetworkInterface) bool { return true })
+
+			ifaces, networks := clearDetachedInterfaces(specIfaces, testNetworks, testStatusIfaces)
+
+			Expect(ifaces).To(Equal(expectedInterfaces))
+			Expect(networks).To(Equal(expectedNetworks))
+		},
+		Entry("should remain, given non-absent interfaces, and no associated status ifaces (i.e.: plug pending)",
+			[]v1.Interface{{Name: "blue"}, {Name: "red"}},
+			[]v1.VirtualMachineInstanceNetworkInterface{},
+			[]v1.Interface{{Name: "blue"}, {Name: "red"}},
+			[]v1.Network{{Name: "blue"}, {Name: "red"}},
+		),
+		Entry("should remain, given non-absent interfaces, and associated status ifaces (i.e.: plugged iface)",
+			[]v1.Interface{{Name: "blue"}, {Name: "red"}},
+			[]v1.VirtualMachineInstanceNetworkInterface{{Name: "blue"}, {Name: "red"}},
+			[]v1.Interface{{Name: "blue"}, {Name: "red"}},
+			[]v1.Network{{Name: "blue"}, {Name: "red"}},
+		),
+		Entry("should remain, given absent iface and associated status ifaces (i.e.: unplug pending)",
+			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}, {Name: "red"}},
+			[]v1.VirtualMachineInstanceNetworkInterface{{Name: "blue"}, {Name: "red"}},
+			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}, {Name: "red"}},
+			[]v1.Network{{Name: "blue"}, {Name: "red"}},
+		),
+		Entry("should be cleared, given absent iface and no associated status iface (i.e.: unplugged iface)",
+			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}, {Name: "red"}},
+			[]v1.VirtualMachineInstanceNetworkInterface{{Name: "red"}},
+			[]v1.Interface{{Name: "red"}},
+			[]v1.Network{{Name: "red"}},
+		),
+		Entry("should remain, given status iface and no associated iface in spec (e.g.: iface reported by GA)",
```

Comment:
I find this unit test redundant and even confusing. GA interface cannot be an input to this method.

Reply:
The "GA" can be removed, but the scenario is valid.
Unit tests are checking a specific unit, and this unit can have such an input. The fact that callers will never use it as such is irrelevant.

At the minimum, you can look at this as documentation of what will happen if such input is given.

Reply:
ok, so let's remove the GA, I find it confusing.

Reply:
Note about GA is removed.

Note that this scenario is valid, for example when there are interfaces in the guest that do not exist in the spec.